powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/persistence/cache/LRUCache.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.idempotency.persistence.cache;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Implementation of a simple LRU Cache based on a {@link LinkedHashMap}
+ * See <a href="https://stackoverflow.com/a/6400874/270653">here</a>.
+ * @param <K> Type of the keys
+ * @param <V> Types of the values
+ */
+public class LRUCache<K, V> extends LinkedHashMap<K, V> {
```

Comment:
Maybe this could be part of the utilities / shared. As an LRU cache might be useful for other features to?

Reply:
Yep, we can probably put it in the utilities. Need to see if we can also reuse it for parameters.